### PR TITLE
Added slippage param, support bsc pancake swap, refactor, fixed swap for input native token

### PR DESCRIPTION
**Pull request by napalmpapalam**

### CODE REVIEW

File: packages/provider/src/provider.ts [hunk 1]
```diff
@@ -59,14 +59,6 @@ export class Provider implements IProvider {
     await this.#proxy.connect()
   }
 
-  public async disconnect() {
```
Comment (ihordiachenko) [inline packages/provider/src/provider.ts:62]: Why did you remove it?

File: packages/nft-checkout/src/operations/evm/helpers/estimator/get-swap-amount.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Price } from '@/entities'
+import { BN } from '@distributedlab/utils'
+
+export const getSwapAmount = (price: Price) => {
+  const amountBN = new BN(price.value).fromFraction(price.decimals)
+
+  const fee = new BN(
+    new BN(amountBN.toString(), { decimals: 24 }).mul(2.5).toString(),
```
Comment (ihordiachenko) [inline packages/nft-checkout/src/operations/evm/helpers/estimator/get-swap-amount.ts:8]: Extract bridge fee into a constant and export it

File: packages/nft-checkout/src/operations/evm/helpers/estimator/joe-trader.ts [hunk 3]
```diff
@@ -12,23 +12,40 @@ import {
...
   TradeType,
 } from '@traderjoe-xyz/sdk'
+import { Price } from '@/entities'
+import { validateSlippage } from './slippage'
+import { getFromToken } from './check-native-token'
 
-export const estimateAvalancheV2 = async (
+const getSlippage = (slippage?: number): Percent => {
+  if (!slippage) {
+    return new Percent('5', '100')
```
Comment (ihordiachenko) [inline packages/nft-checkout/src/operations/evm/helpers/estimator/joe-trader.ts:21]: Move the default slippage into a constant

File: packages/nft-checkout/src/operations/evm/index.ts [hunk 4]
```diff
@@ -101,48 +126,56 @@ export class EVMOperation implements INFTCheckoutOperation {
...
+      isNativeToken(this.#chains, e.from)
+
+    const functionFragment = isV2NativeToken
+      ? 'swapExactNativeInputMultiHopThenBridge'
+      : 'swapExactOutputMultiHopThenBridge'
+
+    return new utils.Interface(
+      SWAP_CONTRACT_ABIS[chain.contactVersion],
+    ).encodeFunctionData(functionFragment, [
+      getSwapAmount(this.#target!.price), // amount out
```
Comment (ihordiachenko) [inline packages/nft-checkout/src/operations/evm/index.ts:168]: Move the arguments to properly named variables and you won't need the comments

File: packages/provider/src/provider.ts (hunk 1 above)
Comment (napalmpapalam) [reply packages/provider/src/provider.ts:62]: coz u cant disconnect the wallet extension from app, it's acquirable only through the extension
